kubernetes/src/main/java/io/kubernetes/client/ApiClient.java

```diff
@@ -949,6 +949,26 @@ public File prepareDownloadFile(Response response) throws IOException {
             return File.createTempFile(prefix, suffix, new File(tempFolderPath));
     }
 
+    public <T> Watch<T> watch(Call call, Type watchType) throws ApiException {
+        try {
+            Response response = call.execute();
+            if (!response.isSuccessful()) {
+                String respBody = null;
+                if (response.body() != null) {
+                    try {
+                        respBody = response.body().string();
+                    } catch (IOException e) {
+                        throw new ApiException(response.message(), e, response.code(), response.headers().toMultimap());
+                    }
+                }
+                throw new ApiException(response.message(), response.code(), response.headers().toMultimap(), respBody);
+            }
+            return new Watch<>(this.json, response.body(), watchType);
+        } catch (IOException e) {
```

Comment:
Why not have this throw IOException also? Seems better than catching and wrapping...